s3-uploads.php

```diff
@@ -21,6 +21,13 @@ function s3_uploads_init() {
 		return;
 	}
 
+	// Make sure the plugin is enabled when autoenable is on, or in a CLI system (so the migrate command works)
+	if ( ( ! defined( 'WP_CLI') || ! WP_CLI ) &&                                      // If CLI is enabled, skip our other checks
```

Comment:
We need to make sure the plugin is "enabled" when CLI is being used, otherwise the migration routines will not fire. I'm not sure how your better approach would work in this case ...